src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -75,7 +75,17 @@ public string FormatValue(object value, string format)
             object routeValues,
             object htmlAttributes)
         {
-            var url = _urlHelper.Action(actionName, controllerName, routeValues, protocol, hostname, fragment);
+            var url = _urlHelper.Action(
+                new UrlActionContext()
+                {
+                    Action = actionName,
+                    Controller = controllerName,
+                    Values = routeValues,
+                    Protocol = protocol,
+                    Host = hostname,
+                    Fragment = fragment
```

Comment:
nit: if you leave a comma at the end, you get less line changes if you add a property. Not that critical here, just a good practice for the future